include/aspect/material_model/simple_nonlinear.h

```diff
@@ -0,0 +1,115 @@
+/*
+  Copyright (C) 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef __aspect__model_simple_nonlinear_h
+#define __aspect__model_simple_nonlinear_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    /**
+     * A material model based on a simple powerlaw rheology and
+     * Implements the derivatives needed for the Newton method.
+     *
+     * Power law equation to compute the viscosity $\eta$ per composition:
+     * $\eta = A * \dot\varepsilon_{II}^{\frac{1}{n}-1}$ where  $A$ is the
+     * prefactor, $\dot\varepsion$ is the strain-rate, II indicates
+     * the square root of the second invariant defined as
+     * $\frac{1}{2} \dot\varepsilon_{ij} \dot\varepsilon_{ij}$, and
+     * $n$ is the stress exponent.
+     *
+     * Afther this the viscosities per composition are averaged using the
+     * utilities weightd p-norm function, where the volume fractions are
+     * the weights.
+     *
+     * @ingroup MaterialModels
+     */
+    template <int dim>
+    class SimpleNonlinear : public MaterialModel::Interface<dim>, public ::aspect::SimulatorAccess<dim>
```

Comment:
What is the reason for that? I could call it power law, which would be more appropriate, but I would also like to make a the distinction that that it is practically the simplest form of a power law possible, and not a power law like diffussion and/or dislocation creep.